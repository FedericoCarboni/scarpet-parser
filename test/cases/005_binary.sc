// Binary Operator precedence test
1 + (2 + 3) * 4 ^ 5 / 2 % 3
