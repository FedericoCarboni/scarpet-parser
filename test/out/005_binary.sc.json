{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "start": {
      "pos": 35,
      "row": 0,
      "col": 0
    },
    "end": {
      "pos": 48,
      "row": 0,
      "col": 0
    },
    "operator": "+",
    "lvalue": {
      "kind": "NumberLiteral",
      "start": {
        "pos": 35,
        "row": 0,
        "col": 0
      },
      "end": {
        "pos": 36,
        "row": 0,
        "col": 0
      },
      "value": "1"
    },
    "rvalue": {
      "kind": "BinaryExpression",
      "start": {
        "pos": 39,
        "row": 0,
        "col": 0
      },
      "end": {
        "pos": 48,
        "row": 0,
        "col": 0
      },
      "operator": "*",
      "lvalue": {
        "kind": "NumberLiteral",
        "start": {
          "pos": 39,
          "row": 0,
          "col": 0
        },
        "end": {
          "pos": 40,
          "row": 0,
          "col": 0
        },
        "value": "2"
      },
      "rvalue": {
        "kind": "BinaryExpression",
        "start": {
          "pos": 43,
          "row": 0,
          "col": 0
        },
        "end": {
          "pos": 48,
          "row": 0,
          "col": 0
        },
        "operator": "^",
        "lvalue": {
          "kind": "NumberLiteral",
          "start": {
            "pos": 43,
            "row": 0,
            "col": 0
          },
          "end": {
            "pos": 44,
            "row": 0,
            "col": 0
          },
          "value": "3"
        },
        "rvalue": {
          "kind": "NumberLiteral",
          "start": {
            "pos": 47,
            "row": 0,
            "col": 0
          },
          "end": {
            "pos": 48,
            "row": 0,
            "col": 0
          },
          "value": "4"
        }
      }
    }
  }
}