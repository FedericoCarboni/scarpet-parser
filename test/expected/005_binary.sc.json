{
    "errors": [],
    "warnings": [],
    "root": {
        "kind": "BinaryExpression",
        "start": {
            "pos": 35,
            "row": 1,
            "col": 0
        },
        "end": {
            "pos": 54,
            "row": 1,
            "col": 19
        },
        "operator": "+",
        "lvalue": {
            "kind": "NumberLiteral",
            "start": {
                "pos": 35,
                "row": 1,
                "col": 0
            },
            "end": {
                "pos": 36,
                "row": 1,
                "col": 1
            },
            "value": "1"
        },
        "rvalue": {
            "kind": "BinaryExpression",
            "start": {
                "pos": 39,
                "row": 1,
                "col": 4
            },
            "end": {
                "pos": 54,
                "row": 1,
                "col": 19
            },
            "operator": "*",
            "lvalue": {
                "kind": "ParenthesisedExpression",
                "start": {
                    "pos": 39,
                    "row": 1,
                    "col": 4
                },
                "end": {
                    "pos": 46,
                    "row": 1,
                    "col": 11
                },
                "value": {
                    "kind": "BinaryExpression",
                    "start": {
                        "pos": 40,
                        "row": 1,
                        "col": 5
                    },
                    "end": {
                        "pos": 45,
                        "row": 1,
                        "col": 10
                    },
                    "operator": "+",
                    "lvalue": {
                        "kind": "NumberLiteral",
                        "start": {
                            "pos": 40,
                            "row": 1,
                            "col": 5
                        },
                        "end": {
                            "pos": 41,
                            "row": 1,
                            "col": 6
                        },
                        "value": "2"
                    },
                    "rvalue": {
                        "kind": "NumberLiteral",
                        "start": {
                            "pos": 44,
                            "row": 1,
                            "col": 9
                        },
                        "end": {
                            "pos": 45,
                            "row": 1,
                            "col": 10
                        },
                        "value": "3"
                    },
                    "comment": "// Binary Operator precedence test\n"
                }
            },
            "rvalue": {
                "kind": "BinaryExpression",
                "start": {
                    "pos": 49,
                    "row": 1,
                    "col": 14
                },
                "end": {
                    "pos": 54,
                    "row": 1,
                    "col": 19
                },
                "operator": "^",
                "lvalue": {
                    "kind": "NumberLiteral",
                    "start": {
                        "pos": 49,
                        "row": 1,
                        "col": 14
                    },
                    "end": {
                        "pos": 50,
                        "row": 1,
                        "col": 15
                    },
                    "value": "4"
                },
                "rvalue": {
                    "kind": "NumberLiteral",
                    "start": {
                        "pos": 53,
                        "row": 1,
                        "col": 18
                    },
                    "end": {
                        "pos": 54,
                        "row": 1,
                        "col": 19
                    },
                    "value": "5"
                },
                "comment": "// Binary Operator precedence test\n"
            },
            "comment": "// Binary Operator precedence test\n"
        },
        "comment": "// Binary Operator precedence test\n"
    }
}